rpc/core/mempool.go

```diff
@@ -137,7 +139,7 @@ func BroadcastTxCommit(ctx *rpctypes.Context, tx types.Tx) (*ctypes.ResultBroadc
 				Hash:      tx.Hash(),
 			}, err
 		case <-time.After(env.Config.TimeoutBroadcastTxCommit):
-			err = errors.New("timed out waiting for tx to be included in a block")
+			err = ErrTimedOutWaitingForTx
```

Comment:
[Suggestion] While there has been a particular interest in defining this error as a named type, I believe it would be more effective and consistent to refactor the enclosing function so that all returned errors are named errors.

Reply:
I agree it would be consistent to export more errors but idk if it would be more effective b/c no downstream consumers have concrete plans to use any other errors. Exporing more errors here increases the divergence between this repo and upstream so I think there's a trade-off.

Reply:
> Exporing more errors here increases the divergence between this repo and upstream so I think there's a trade-off. 

I see, I agree that keeping the differences minimal might be a good approach as well.  Nevertheless, if further named errors are to be added, upstreaming the changes could be an option as well.

Reply:
Agreed. Created https://github.com/cometbft/cometbft/issues/2014